.travis.yml

```diff
@@ -23,8 +23,6 @@ matrix:
     - jdk: openjdk8
     - jdk: openjdk11
     - jdk: openjdk11
-      env:
-        - SLOW_TESTS=Y
```

Comment:
Do you intend to drop Travis tests completely? It looks like job 2 and 3 duplicate each other now, and job3 was testing the slow tests earlier.

Reply:
I would prefer to remove them completely but I've seen that you wanted to keep them around for a bit so I am fine either way.

Reply:
Could you please keep it for a while (log viewer is really better at Travis), but I guess it might make sense to configure it to run the slow tests rather than duplicate existing openjdk11 tests

Reply:
OK for leaving one config in Travis but I don't think slow tests can remain there. Slow tests need around 1h 20m to run and some of them require quite a bit of memory (Xmx6g). Travis has time and memory limitations so in this case it cannot be used. For those reasons I think it is better to run them using actions (https://github.com/zabetak/calcite/runs/314646481).

Reply:
If the test takes 1h20m, then I would suggest running the test for `master` branch only once a day.
Or once per `master` commit.

Running 1h20m tests does not sound a fair use of GitHub resources, and I do not really think PRs need to wait 1h20m in order to get feedback.

Reply:
This will be run as the last action so if somebody wants to give feedback before (without waiting the slow tests) it will be possible. Anyways there are not many PRs who receive feedback in less than an hour. 

In terms of resources yes I agree that it is not great but shouldn't we let GitHub worry about this. Given that at the moment we don't have restrictions shouldn't we try to control better what is going in the master. 

Other than that I am fine enabling the action only per master commit.

Reply:
>Anyways there are not many PRs who receive feedback in less than an hour.

Regular tests complete within 20min, which is well under 1h.

> In terms of resources yes I agree that it is not great but shouldn't we let GitHub worry about this

This is a chicken-and-egg problem: Travis has a limited set of executors for all ASF projects, so running long tests does impact other projects.
See https://lists.apache.org/thread.html/af52e2a3e865c01596d46374e8b294f2740587dbd59d85e132429b6c@%3Cbuilds.apache.org%3E
In the same way, if other projects could reduce the waste, that would improve "queueing" times for Calcite as well.

Even though some `slow` tests might be important, I suspect that the notable part of the `slow` tests does nothing but burning CPU (all of them?).
For instance, in https://github.com/apache/calcite/commit/f2f25b3a567ee7ec1c6e8f9f00f581b2dacc3dae I "reduced" the heap consumption of `FoodmartTest` by 400MiB. Does that mean the test was using the memory for a good reason? I doubt so. I think it was just a bug.

It is clear that launching `slow` tests for each and every commit would burn a lot of CPU, and the gain would be invisible.

Reply:
OK agreed! Let's run them only on push to master!